static/app/bootstrap/bootstrapRequests.tsx

```
@@ -0,0 +1,200 @@
+
+// 10 minute gc time
+// gc time is how long it will stay (even if stale) in the query cache
+const BOOTSTRAP_QUERY_GC_TIME = 10 * 60 * 1000;
+
+export function useBootstrapOrganizationQuery(orgSlug: string | null) {
+  const organizationQuery = useQuery(getBootstrapOrganizationQueryOptions(orgSlug));
+
+  useEffect(() => {
+    if (organizationQuery.data) {
+      OrganizationStore.onUpdate(organizationQuery.data);
+    }
+    if (organizationQuery.error) {
+      OrganizationStore.onFetchOrgError(organizationQuery.error as any);
+    }
+  }, [organizationQuery.data, organizationQuery.error]);
+
+  return organizationQuery;
+}
+
+export function useBootstrapTeamsQuery(orgSlug: string | null) {
+  const teamsQuery = useQuery(getBoostrapTeamsQueryOptions(orgSlug));
+
+  useEffect(() => {
+    if (teamsQuery.data) {
+      TeamStore.loadInitialData(
+        teamsQuery.data.teams,
+        teamsQuery.data.hasMore,
+        teamsQuery.data.cursor
+      );
+    }
+  }, [teamsQuery.data]);
+
+  return teamsQuery;
+}
+
+export function useBootstrapProjectsQuery(orgSlug: string | null) {
+  const projectsQuery = useQuery(getBootstrapProjectsQueryOptions(orgSlug));
+
+  useEffect(() => {
+    if (projectsQuery.data) {
+      ProjectsStore.loadInitialData(projectsQuery.data);
+    }
+  }, [projectsQuery.data]);
+
+  return projectsQuery;
+}
+
+export function getBootstrapOrganizationQueryOptions(orgSlug: string | null) {
+  return queryOptions({
+    queryKey: ['bootstrap-organization', orgSlug],
+    queryFn: async (): Promise<Organization> => {
+      const uncancelableApi = new Client();
+      // Get the preloaded data promise
+      try {
+        const preloadResponse = await getPreloadedData('organization', orgSlug!);
+        // If the preload request was for a different org or the promise was rejected
+        if (Array.isArray(preloadResponse) && preloadResponse[0] !== null) {
+          return preloadResponse[0];
+        }
+      } catch {
+        //
```

since this is a critical app path, we should report the exception to sentry (if its noisy, we can always remove it in the future)